limiter/limiter.go

```diff
@@ -24,11 +25,15 @@ func New(maxRequests, rps int, maxDuration time.Duration) *Run {
 	}
 
 	r := &Run{
+		unlimited:    *atomic.NewBool(maxRequests == 0),
```

Comment:
Does this actually need to be an atomic? It only changes during stop, and a non-atomic read then should be fine.